system_metrics_collector/src/system_metrics_collector/utilities.cpp

```diff
@@ -48,6 +49,108 @@ std::string readFileToString(const std::string & file_name)
   return to_return;
 }
 
+ProcCpuData processStatCpuLine(const std::string & stat_cpu_line)
+{
+  ProcCpuData parsed_data;
+
+  if (!stat_cpu_line.empty()) {
+    if (!stat_cpu_line.compare(0, strlen(CPU_LABEL), CPU_LABEL)) {
+      std::istringstream ss(stat_cpu_line);
+
+      if (!ss.good()) {
+        return ProcCpuData();
+      }
+      ss >> parsed_data.cpu_label;
+
+      for (int i = 0;
+        i < static_cast<int>(ProcCpuStates::kNumProcCpuStates); ++i)
+      {
+        if (!ss.good()) {
+          return ProcCpuData();
+        }
+        ss >> parsed_data.times[i];
+      }
+      return parsed_data;
+    }
+  }
+  return parsed_data;
+}
+
+ProcPidCpuData processPidStatCpuLine(const std::string & stat_cpu_line)
+{
+  ProcPidCpuData parsed_data;
+
+  if (!stat_cpu_line.empty()) {
+    std::istringstream ss(stat_cpu_line);
+
+    ss.ignore();  // pid
+    ss.ignore();  // comm
+    ss.ignore();  // state
+    ss.ignore();  // ppid
+    ss.ignore();  // pgrp
+    ss.ignore();  // session
+    ss.ignore();  // tty_nr
+    ss.ignore();  // tpgid
+    ss.ignore();  // flags
+    ss.ignore();  // minflt
+    ss.ignore();  // cminflt
+    ss.ignore();  // majflt
+    ss.ignore();  // cmajflt
+    ss >> parsed_data.utime;
```

Comment:
Don't you need to check if good after all the operations? You could set the failbits to throw as suggested in another PR that I had to change.

Reply:
https://en.cppreference.com/w/cpp/io/basic_ios/exceptions

Reply:
Done.